app.rb

```diff
@@ -0,0 +1,126 @@
+require_relative 'book'
+require_relative 'person'
+require_relative 'rental'
+require_relative 'student'
+require_relative 'teacher'
+
+class App
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  # Create a book
+  def create_book
+    print 'Whats the book title?:'
+    title = gets.chomp
+    print 'Who is the author?:'
+    author = gets.chomp
+    book = Book.new(title, author)
+    @books.push(book)
+    puts 'Awesome! Book created successfully'
+  end
+
+  # List all books
+  def list_books
+    if @books.empty?
+      puts 'Aww, no books found'
+    else
+      @books.each do |book|
+        puts "#{book.title} by #{book.author}"
+      end
+    end
+  end
+
+  # List people
+  def list_people
+    if @people.empty?
+      puts 'Hmm, no person found'
+    else
+      @people.each do |person|
+        puts "Name: #{person.name} | Age: #{person.age} | ID: #{person.id}"
+      end
+    end
+  end
+
+  # Create student
+  def create_student
+    print 'What\'s the age of the new student?:'
+    age = gets.chomp
+    print 'What\'s the name?:'
+    name = gets.chomp
+    print 'Should parent permission be assigned? [Y/N]:'
+    parent_permission = gets.chomp.downcase
+    case parent_permission
+    when 'n'
+      student = Student.new(age: age, classroom: nil, name: name, parent_permission: false)
+      @people.push(student)
+      puts 'Awesome! Student created successfully'
+    when 'y'
+      student = Student.new(age: age, classroom: nil, name: name, parent_permission: true)
+      @people.push(student)
+      puts 'Awesome! Student created successfully'
+    end
+  end
+
+  # Create teacher
+  def create_teacher
+    print 'What\'s the age of the new teacher?:'
+    age = gets.chomp
+    print 'What about the name:'
+    name = gets.chomp
+    print 'What\'s the specialization?:'
+    specialization = gets.chomp
+    teacher = Teacher.new(age: age, specialization: specialization, name: name)
+    @people.push(teacher)
+    puts 'Awesome! Teacher created successfully'
+  end
+
+  # Create person
+  def create_person
+    print 'Start by selecting Student (1) or Teacher (2):'
+    person_type = gets.chomp
+    case person_type
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      puts 'Invalid option. Please select 1 or 2'
+    end
+  end
```

Comment:
- [ ] When I'm trying to create a person (student or teacher), I'm getting the below error. Please make sure to fix this issue and users should be able to create a person successfully. After fixing this issue also make sure to creating and reading the rentals also works as expected.

![image](https://user-images.githubusercontent.com/83803414/219879833-8a6de870-6ccb-4743-9b27-a3df9e7bdb57.png)